lib/rules-generator.js

```diff
@@ -366,9 +377,31 @@ methods(Generator, {
 
   // Lookup globally defined function.
   lookupFunction: function(ref) {
-    return ref.type == 'var' && this.symbols.functions[ref.name];
+    if (ref.type == 'var') {
+      var fn = this.symbols.functions[ref.name];
+      if (!fn) {
+        this.fatal("Undefined global function: " + ref.name);
+      }
+      return fn;
+    }
+    // TODO: Validate "method" calls against a prototype.
+    return undefined;
   },
 
+  silent: function(f) {
+    if (f === undefined) {
+      f = true;
+    }
+    this.runSilently = f;
```

Comment:
I didn't indent to "clean" a truthy value here - just to allow for optional argument.